scripts/isolate_tests.py

```diff
@@ -35,54 +36,44 @@ def extract_test_cases(path):
 
     return tests
 
-# Contract sources are indented by 4 spaces.
-# Look for `pragma solidity`, `contract`, `library` or `interface`
-# and abort a line not indented properly.
+# Extract code examples based on a start marker
+# up until we reach EOF or a line that is not empty and doesn't start with 4
+# spaces.
 def extract_docs_cases(path):
+    beginMarkers = ['.. code-block:: solidity', '::']
+    immediatelyAfterMarker = False
     insideBlock = False
-    insideBlockParameters = False
-    pastBlockParameters = False
-    extractedLines = []
     tests = []
 
     # Collect all snippets of indented blocks
-
     with open(path, mode='r', errors='ignore', encoding='utf8', newline='') as f:
         lines = f.read().splitlines()
-    for l in lines:
-        if l != '':
-            if not insideBlock and l.startswith(' '):
-                # start new test
-                extractedLines += ['']
-                insideBlockParameters = False
-                pastBlockParameters = False
-            insideBlock = l.startswith(' ')
+
+    for line in lines:
         if insideBlock:
-            if not pastBlockParameters:
-                # NOTE: For simplicity this allows blank lines between block parameters even
-                # though Sphinx does not. This does not matter since the first non-empty line in
-                # a Solidity file cannot start with a colon anyway.
-                if not l.strip().startswith(':') and (l != '' or not insideBlockParameters):
-                    insideBlockParameters = False
-                    pastBlockParameters = True
-                else:
-                    insideBlockParameters = True
-
-            if not insideBlockParameters:
-                extractedLines[-1] += l + '\n'
+            if immediatelyAfterMarker:
+                # Skip Shpinx instructions and empty lines between them
+                if line == '' or line.lstrip().startswith(":"):
+                    continue
+
+            if line == '' or line.startswith("    "):
+                tests[-1] += line + "\n"
+                immediatelyAfterMarker = False
+            else:
+                insideBlock = False
+        elif any(map(line.lower().startswith, beginMarkers)):
+            insideBlock = True
+            immediatelyAfterMarker = True
+            tests += ['']
 
     codeStart = "(// SPDX-License-Identifier:|pragma solidity|contract.*{|library.*{|interface.*{)"
 
-    # Filter all tests that do not contain Solidity or are indented incorrectly.
-    for lines in extractedLines:
-        if re.search(r'^\s{0,3}' + codeStart, lines, re.MULTILINE):
-            print("Indentation error in " + path + ":")
-            print(lines)
-            exit(1)
```

Comment:
I think this message was useful even if it might be annoyoing at times. Without it if you make a mistake and use tabs or 2 spaces to indent, the snippet will just be silently skipped and no one will notice.